### PR TITLE
zereox settler - base 

### PR DESCRIPTION
**Pull request by RantumBits**
### Contribution type
Please check the type of contribution this pull request is for:

- [x] New spell(s)
- [ ] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### CODE REVIEW

**Comment by RantumBits** [comment]
@jeff-dude I wasn't able to recreate the error on the frontend but did make some changes to `token_sold_amount_raw` and `token_bought_amount_raw` .

**Comment by jeff-dude** [comment]
> @jeff-dude I wasn't able to recreate the error on the frontend but did make some changes to `token_sold_amount_raw` and `token_bought_amount_raw` here

can you force changes to models downstream, so CI runs and try to recreate error?
should be up to `zeroex.trades`, `dex_aggregator.trades`

**Comment by RantumBits** [comment]
> > @jeff-dude I wasn't able to recreate the error on the frontend but did make some changes to `token_sold_amount_raw` and `token_bought_amount_raw` here
> 
> can you force changes to models downstream, so CI runs and try to recreate error? should be up to `zeroex.trades`, `dex_aggregator.trades`

all set now with downstream merging @jeff-dude

**Comment by RantumBits** [comment]
> > @jeff-dude I wasn't able to recreate the error on the frontend but did make some changes to `token_sold_amount_raw` and `token_bought_amount_raw` here
> 
> can you force changes to models downstream, so CI runs and try to recreate error? should be up to `zeroex.trades`, `dex_aggregator.trades`

all set now with downstream merging @jeff-dude

**Comment by jeff-dude** [comment]
> all set now with downstream merging @jeff-dude

nice!

is it possible to close all other related PRs and open one that reverts [this](https://github.com/duneanalytics/spellbook/commit/4c597469657cd714afff675d4bf4ec538ff9e163) commit? that way it's all back in one PR and clean to merge.

then, also if you don't mind, can you comment inline of code of the new PR where the fix is so i can be aware?

**Comment by RantumBits** [comment]
> > all set now with downstream merging @jeff-dude
> 
> nice!
> 
> is it possible to close all other related PRs and open one that reverts [this](https://github.com/duneanalytics/spellbook/commit/4c597469657cd714afff675d4bf4ec538ff9e163) commit (basically revert my revert)? that way it's all back in one PR and clean to merge.
> 
> then, also if you don't mind, can you comment inline of code of the new PR where the fix is so i can be aware?

not quite fixed yet @jeff-dude. that last successful run didn't include a run of `zeroex_trades`. When `zeroex_trades` runs , `dex_aggregator.trades` fails. When running only `dex_aggregator.trades`, it's successful. 

I am checking into this further

**Comment by RantumBits** [comment]
@jeff-dude looks like this may be due to another issue. There are 3 existing swaps in dex_aggregator.trades with the value of 115792089237316195423570985008687907853269984665640564039457584007913129639935

check this - https://dune.com/queries/3958561

**Comment by jeff-dude** [comment]
> @jeff-dude looks like this may be due to another issue. There are 3 existing swaps in dex_aggregator.trades with the value of 115792089237316195423570985008687907853269984665640564039457584007913129639935
> 
> check this - https://dune.com/queries/3958561

ohhh interesting, good find. i will have to dig a bit.

also, side note, i am changing CI cluster size so if you get random failures, that's me

File: dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql [hunk 1]
```diff
@@ -44,8 +44,8 @@ SELECT
     , token_pair
     , token_bought_amount
     , token_sold_amount
-    , token_bought_amount_raw
-    , token_sold_amount_raw
+    , try_cast(token_bought_amount_raw as int256) as token_bought_amount_raw
+    , try_cast(token_sold_amount_raw as int256) as token_sold_amount_raw
```
Comment (RantumBits) [inline dbt_subprojects/dex/models/aggregator_trades/dex_aggregator_trades.sql:48]: @jeff-dude raw_amount issue is resolved with a try_cast in `dex_aggregator_trades`. Are you ok with this approach?

**Comment by RantumBits** [comment]
@jeff-dude there's a separate txn failing this test: accepted_range_dex_aggregator_trades_amount_usd__1000000000 
see:  `select * from hive.test_schema.git_dunesql_4e703e3_dbt_utils_accepted_range_dex_aggregator_trades_amount_usd__1000000000`

**Comment by jeff-dude** [comment]
> @jeff-dude there's a separate txn failing this test: accepted_range_dex_aggregator_trades_amount_usd__1000000000
> 
> `select * from hive.test_schema.git_dunesql_4e703e3_dbt_utils_accepted_range_dex_aggregator_trades_amount_usd__1000000000` returns tx_hash 0x22239a0d02c2975b71b290e28b35723bdc8cf5be76ff7189d1ef6a8e62a79ed2

i just got the same failure in prod, so let me try to fix that, then we can revisit your above question

**Comment by jeff-dude** [comment]
closing this one so we can focus on new with all chains

**Pull request closed without merging**